cdk/dcm_floodlight/transform.js

```diff
@@ -19,14 +19,17 @@ function appendProperties(endpoint, payload) {
 
 // transform webapp dynamicForm custom floodlight variable
 // into {u1: value, u2: value, ...}
-function transformCustomVariable(customFloodlightVariable) {
+function transformCustomVariable(customFloodlightVariable, message) {
   const customVariable = {};
   customFloodlightVariable.forEach(item => {
     if (item && !isEmpty(item.from) && !isEmpty(item.to)) {
       // remove u if already there
-      customVariable[
-        `u${item.from.trim().replace(/u/g, "")}`
-      ] = `[${item.to.trim()}]`;
+      if (get(message, `properties.${item.to.trim()}`)) {
```

Comment:
```suggestion
      const propertyItem = get(message, `properties.${item.to.trim()}`);
      if (propertyItem) {
        // Not sure how important the casting to string is here
        customVariable[`u${item.from.trim().replace(/u/g, "")}`] = `${propertyItem}`;
      }
```
Can we change the implementation to this ?

Reply:
One more thing which competition supports is is json style dot notation accessors wrapped as handle bar expressions example {{ context.campaign.name}} how much of an effor is that for supporting that we can use handlebar expressions

Reply:
We can re-use the dynamic config module for this.

Reply:
We already support handlebars way of implementation in case of mappings as well. You can check it in `[handleMetadataForValue](https://github.com/rudderlabs/rudder-transformer/blob/master/v0/util/index.js#L556-L559)` method. That should give us a lot more flexibility. If you meant something else, let me know

Reply:
@sanpj2292 can this be done in post mapper? Because this metadata is part of construct payload

Reply:
@utsabc You can import `handlebars` or `set-value` into `transform.js` & implement it

Reply:
Got it @akashrpo lets do it

Reply:
@saikumarrs we want this feature in device mode as ewll no technical blockers there?

Reply:
As discussed added the support to take the values first from `properties` and if not found we are taking from root level `message.*`